test/unit/mdc-textfield/foundation.test.js

```diff
@@ -502,6 +481,36 @@ test('on focus makes helper text visible to the screen reader', () => {
   td.verify(helperText.showToScreenReader());
 });
 
+test('on required validation attribute change should update invalid classes', () => {
```

Comment:
I'm confused why these 3 tests were added when the 2 below were also added. Aren't both sets testing the same thing? The ones below seem more appropriate.

Reply:
True, these are basically duplicates while the 2 below handle the general case. I'll remove.